### PR TITLE
update version vfx forward movement

### PR DESCRIPTION
**Pull request by AlecatEnjoyer**
jira: https://bradleycapstone.atlassian.net/browse/LV-512

this branch includes the fix to the smoke vfx bug

### CODE REVIEW

File: Assets/General/Scripts/Health/GeneralVfxFunctionality.cs [hunk 1]
```diff
@@ -46,7 +50,17 @@ public void StartAllVfx()
...
     }
-
+/// <summary>
+/// To move Vfx forward if need be
+/// </summary>
+    void Update()
+    {
+        if (_isMoveableVfx)
+        {
+            transform.position += ((transform.forward /2f) * Time.deltaTime);
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Health/GeneralVfxFunctionality.cs:60]: Perhaps speed should be modifiable.

File: Assets/General/Scripts/Health/GeneralVfxFunctionality.cs [hunk 2]
```diff
@@ -46,7 +50,17 @@ public void StartAllVfx()
             animator.SetTrigger(VFX_ANIMATION_TRIGGER);
         }
     }
-
+/// <summary>
+/// To move Vfx forward if need be
+/// </summary>
+    void Update()
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Health/GeneralVfxFunctionality.cs:56]: Personally would prefer this as a coroutine, but I can't force ya

File: Assets/General/Scripts/Health/GeneralVfxFunctionality.cs [hunk 3]
```diff
@@ -6,6 +6,8 @@
 // Description:     Handles any general functionality related to an object with particle systems
 ******************************************************************************/
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Health/GeneralVfxFunctionality.cs:7]: I see this file is located under Assets/General/Scripts/Health. Not sure why it's under health. Maybe I did that I have no idea, I know it wasn't your fault here.

**Review by RyanASwanson — APPROVED** [review]
Left some suggestions but I ain't blocking this when we are this close to a deadline

File: Assets/General/Scripts/Health/GeneralVfxFunctionality.cs (hunk 2 above)
Comment (AlecatEnjoyer) [reply Assets/General/Scripts/Health/GeneralVfxFunctionality.cs:56]: I tried coroutine at first but it kept exploding so I gave up on it

File: Assets/General/Scripts/Health/GeneralVfxFunctionality.cs (hunk 1 above)
Comment (AlecatEnjoyer) [reply Assets/General/Scripts/Health/GeneralVfxFunctionality.cs:60]: bet

File: Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs [hunk 4]
```diff
@@ -113,6 +113,8 @@ public enum EFocusState
 
     private PlayerReticle _reticle;
 
+    private Transform _smokeMovement;
```
Comment (ASLH68) [inline Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonGun.cs:116]: Add a tooltip

File: Assets/General/Scripts/Health/GeneralVfxFunctionality.cs [hunk 5]
```diff
@@ -46,7 +54,17 @@ public void StartAllVfx()
             animator.SetTrigger(VFX_ANIMATION_TRIGGER);
         }
     }
-
+/// <summary>
+/// To move Vfx forward if need be
+/// </summary>
+    void Update()
```
Comment (ASLH68) [inline Assets/General/Scripts/Health/GeneralVfxFunctionality.cs:60]: private specifier

**Review by ASLH68 — APPROVED** [review]

**Pull request merged**